server/lib/s3backup/__init__.py

```diff
@@ -0,0 +1,176 @@
+"""
+Simple module level convenience functions.
+"""
+import boto3
+import os
+import sys
+import glob
+import base64
+import hashlib
+import shutil
+import time
+import configtools
+from datetime import datetime
+from configparser import ConfigParser
+
+class Entry(object):
+    """
+    A simple class to create an entry as an object.
+    """
+    def __init__(self, name, md5):
+        self.name = name
+        self.md5 = md5
+
+    def __eq__(self, other):
+        return self.name == other.name and self.md5 == other.md5
+
+
+class S3Obj(object):
+    def __init__(self, s3connector):
+        self.s3connector = s3connector
+
+    def ts(self, *args):
+        return time.strftime("%Y-%m-%dT%H:%M:%S-%Z", time.localtime())
+
+    def list_buckets(self):
+        try:
+            list_bucket = self.s3connector.list_buckets()
+            buckets = [bucket['Name'] for bucket in list_bucket['Buckets']]
+            return buckets
+        except Exception:
+            return []
+
+    def list_objects_v2(self, Bucket):
+        return self.s3connector.list_objects_v2(Bucket=Bucket)
+
+    def put_object(self, Bucket, Key, Body, ContentMD5):
+        return self.s3connector.put_object(Bucket=Bucket, Key=Key,
+                Body=Body, ContentMD5=ContentMD5)
+
+
+class MockS3Obj(object):
```

Comment:
Maybe rename `S3Obj` to `S3Connector` and `MockS3Obj` to `MockS3Connector`? `Obj` is not very descriptive.